### DISCUSSION TITLE
TypeError: Client.put() got an unexpected keyword argument 'extensions'

### DISCUSSION THREAD

**Discussion by aleksandrpahomov** [fastapi/fastapi#6189] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import pytest
from fastapi.testclient import TestClient

@pytest.mark.parametrize(
    ["token", "data", "status_code_returned"],
    [
        pytest.param(USER_ID_CORRECT_TOKEN, UPDATE_DATA, 200, id="update user info success"),
        pytest.param(USER_ID_FAILED_TOKEN, UPDATE_DATA, 404, id="update user info failed"),
    ],
)

def test_update_user_info(client: TestClient, token, data, status_code_returned: int):
    headers = {"Authorization": f"Bearer {token}"}
    response = client.put("/admin/user", json=data, headers=headers)
    assert response.status_code == status_code_returned
    if token == USER_ID_CORRECT_TOKEN:
        assert "success" in response.text
    elif token == USER_ID_FAILED_TOKEN:
        assert "UserNotFoundException" in response.text
```

### Description

Hello! My test worked correctly until I wasn't updated fast api version after commit  f92f87d379cc5d3c9b3c159d74639411617cb0da. Commit  message in Fast API repo was  **"Update references to Requests for tests to HTTPX, and add HTTPX to extras"**. After it, it was always occured an error:

` TypeError: Client.put() got an unexpected keyword argument 'extensions'`

If we check  fastapi **TestClient.put()** method, we will see that we have an "extensions" here. Also we see that **super().put(..)** method called

```
def put(  
        self,
        url: httpx._types.URLTypes,
        *,
        content: httpx._types.RequestContent = None,
        data: httpx._types.RequestData = None,
        files: httpx._types.RequestFiles = None,
        json: typing.Any = None,
        params: httpx._types.QueryParamTypes = None,
        headers: httpx._types.HeaderTypes = None,
        cookies: httpx._types.CookieTypes = None,
        auth: typing.Union[
            httpx._types.AuthTypes, httpx._client.UseClientDefault
        ] = httpx._client.USE_CLIENT_DEFAULT,
        follow_redirects: bool = None,
        allow_redirects: bool = None,
        timeout: typing.Union[
            httpx._client.TimeoutTypes, httpx._client.UseClientDefault
        ] = httpx._client.USE_CLIENT_DEFAULT,
        extensions: dict = None,
    ) -> httpx.Response:
        redirect = self._choose_redirect_arg(follow_redirects, allow_redirects)
        return super().put(
            url,
            content=content,
            data=data,
            files=files,
            json=json,
            params=params,
            headers=headers,
            cookies=cookies,
            auth=auth,
            follow_redirects=redirect,
            timeout=timeout,
            extensions=extensions,
        )
```

BUT! **super().put(..)** method its a **httpx.put()** method and its dont have **extensions** param, let's see source code of httpx 

```
def put(self,
        url: URLTypes,
        *,
        content: RequestContent = None,
        data: RequestData = None,
        files: RequestFiles = None,
        json: typing.Any = None,
        params: QueryParamTypes = None,
        headers: HeaderTypes = None,
        cookies: CookieTypes = None,
        auth: typing.Union[AuthTypes, UseClientDefault] = USE_CLIENT_DEFAULT,
        follow_redirects: typing.Union[bool, UseClientDefault] = USE_CLIENT_DEFAULT,
        timeout: typing.Union[TimeoutTypes, UseClientDefault] = USE_CLIENT_DEFAULT,
    ) -> Response:
```

I'm not totally sure, but it's looks like a bug.
Thank was answering! 

### Operating System

Linux

### FastAPI Version

0.87.0

### Python Version

3.10

**Comment by mrlonis** [comment]
Same issue over here. I have had to downgrade fastapi until something is worked out. The `bump-testclient` did absolutely nothing for me: https://github.com/Kludex/bump-testclient

**Comment by tiangolo** [comment]
Please check and tell me which versions you have, for FastAPI, Starlette, and HTTPX, you can get that with `python -m pip list`.

And make sure you do it on the virtual environment that has your project.

There's also a chance you already had HTTPX installed but an old version, so you might want to update that too.

**Comment by Kludex** [comment]
You need to have `httpx` >= 0.22.0. Otherwise, you can expect it to fail.

@tiangolo https://github.com/Kludex/bump-testclient/pull/5 :man_shrugging:

**Comment by tiangolo** [comment]
Thanks @Kludex ! 🚀

**Comment by aleksandrpahomov** [comment]
@Kludex  It doesn't work for me.

**Comment by aleksandrpahomov** [comment]
@tiangolo I have 
 `fastapi==0.87.0` 
 `httpx== 1.0.0b0`
`starlette==0.21.0`

**Comment by Kludex** [comment, marked as answer]
Install httpx 0.23.0, please.

**Reply by pedzed** [reply]
This should be documented somewhere. Even in 2024 this is still a problem.

Installing 0.27.0 worked for me.

**Comment by aleksandrpahomov** [comment]
@Kludex It works, thanks a lot!

**Comment by ndeebook** [comment]
I have a similar problem: `TypeError: TestClient.get() got an unexpected keyword argument 'json'`
httpx 0.23.0 on my venv did not fix it for me

**Reply by Kludex** [reply]
It's because `TestClient.get()` doesn't accept `json`, use `TestClient.request("GET", ...)` instead.
